### ISSUE TITLE
Debugging .gba with arm-none-eabi-gdb: "Cannot execute this command without a live selected thread."

### ISSUE THREAD

**Issue by HorstBaerbel** [visualboyadvance-m/visualboyadvance-m#181]
I have tried making debugging in VBA-M work from VS Code with the native debugging extension to debug a GBA project. It did not work, so I tried gdb from the command line, but still no dice. It seems the connection works, but gbd and VBA-M do not work together. I haven't used gbd much, so if I've missed something, please tell me...

### Expected behavior
GDB steps through the executable

### Actual behavior
Error message: "Cannot execute this command without a live selected thread."

### Steps to reproduce the behavior
- Start VBA-M. Set GBD->"Break on load"
- Load .gba -> waits for connection
- start gdb from devkitPro (arm-none-eabi-gdb) with .elf source of .gba
- "target remote :55555"
- gbd connects, VBA-M connection dialog closes
- try to "step" or "continue" from gbd 

Now gbd shows: "Cannot execute this command without a live selected thread." and nothing happens in VBA-M. The same error message comes up in VS Code, so I hope I'm not doing it so wrong after all... :/

Which games are affected by the issue (please state if GBA or GB game):
GBA rom which otherwise runs fine in VBA-M.

### Build details

Operating System (Windows, Mac, Linux (state distribution), etc.):
Ubuntu 16.04.03 x64

Version of code (pre-built binary version, or commit ref, or just "master"):
VBA-M from Ubuntu repo, 2.0.1-r201801190125-3e052c9-22~ubuntu16.04.1 amd64

**Comment by felixjones** [comment]
Just want to chime in that I'm also having the very same problem (same steps, same error message) with VBA-M GDB on macOS and Windows.

There doesn't seem to be any live threads in GDB.

**Comment by crawfxrd** [comment]
I am using `visualboyadvance-m-sdl` on Fedora.

---

Using `arm-none-eabi-gdb` from devkitARM r52-1 does **not** work.
```
$ /opt/devkitpro/devkitARM/bin/arm-none-eabi-gdb --version
GNU gdb (GDB) 8.2.1
```
Output from `vbam` shows:
```
Listening for a connection at port 2345
Got a connection from 127.0.0.1 49914
Color depth: 32
Unknown packet vMustReplyEmpty
```
Output from GDB shows:
```
(gdb) target remote :2345
Remote debugging using :2345
(gdb) cont
Cannot execute this command without a live selected thread.
(gdb)
```

---

Using `arm-none-eabi-gdb` from Fedora 29 repo _does_ work.
```
$ arm-none-eabi-gdb -v
GNU gdb (GDB) 7.6.2
```
Output from `vbam` shows:
```
Listening for a connection at port 2345
Got a connection from 127.0.0.1 49954
Color depth: 32
Unknown packet vCont?
```
Output from GDB shows:
```
(gdb) target remote :2345
Remote debugging using :2345
0x08000000 in _start ()
(gdb) cont
Continuing.
```

**Comment by rkitover** [comment]
That's interesting, something must have changed in the gdb protocol between 7 and 8.

**Comment by crawfxrd** [comment]
I tried local builds of v2.1.1 and a8b386816a78da68343c016c9c4c515f7d2e80d2. These both worked with devkitARM r52-1.
```
Listening for a connection at port 2345
Got a connection from 127.0.0.1 50864
Color depth: 32
Unknown packet vMustReplyEmpty
Unknown packet qTStatus
Unknown packet qOffsets
Unknown packet vCont?
```
```
(gdb) target remote :2345
Remote debugging using :2345
0x08000000 in _start ()
(gdb) cont
Continuing.
```

So my issue is that Fedora does not have up-to-date packages.

**Comment by HorstBaerbel** [comment]
Does anyone of you have a working launch.json for debugging in Visual Studio with VBA-m / GDB? I can't seem to make it work...

**Comment by rkitover** [comment]
There were some gdb remote fixes from @knightsc between v2.1.0 and v2.1.1, so this will explain your results.

**Comment by rkitover** [comment]
@Mystro256 hey are you still the fedora maint?
